src/main/java/org/openrewrite/java/spring/boot2/WebSecurityConfigurerAdapterRecipe.java

```diff
@@ -0,0 +1,230 @@
+            }
+
+            @Override
+            public J.Block visitBlock(J.Block block, ExecutionContext context) {
+                J.Block b = super.visitBlock(block, context);
+                if (getCursor().getParent() != null && getCursor().getParent().getValue() instanceof J.MethodDeclaration) {
+                    J.MethodDeclaration parentMethod = getCursor().getParent().getValue();
+                    Cursor classDeclCursor = getCursor().dropParentUntil(J.ClassDeclaration.class::isInstance);
+                    J.ClassDeclaration classDecl = classDeclCursor.getValue();
+                    if (classDeclCursor.getMessage(MSG_QUALIFIES, false)) {
+                        if (CONFIGURE_HTTP_SECURITY_METHOD_MATCHER.matches(parentMethod, classDecl)) {
+                            JavaTemplate template = JavaTemplate.builder(this::getCursor,  "return #{any(org.springframework.security.config.annotation.SecurityBuilder)}.build();")
+                                    .javaParser(() -> JavaParser.fromJavaVersion()
+                                            .dependsOn("package org.springframework.security.config.annotation;" +
+                                                    "public interface SecurityBuilder<O> {\n" +
+                                                    "    O build() throws Exception;" +
+                                                    "}")
+                                            .build()).imports("org.springframework.security.config.annotation.SecurityBuilder").build();
+                            b = b.withTemplate(template, b.getCoordinates().lastStatement(),
+                                    ((J.VariableDeclarations)parentMethod.getParameters().get(0)).getVariables().get(0).getName());
+                        } else if (CONFIGURE_WEB_SECURITY_METHOD_MATCHER.matches(parentMethod, classDecl)) {
+                            J.Lambda.Parameters parameters = new J.Lambda.Parameters(Tree.randomId(), Space.EMPTY, Markers.EMPTY, true, parentMethod.getParameters().stream().filter(J.VariableDeclarations.class::isInstance).map(J.VariableDeclarations.class::cast).map(p -> {
+                                J.VariableDeclarations.NamedVariable varDecl = p.getVariables().get(0);
+                                return new JRightPadded<J>(varDecl.getName(), Space.EMPTY, Markers.EMPTY);
+                            }).collect(Collectors.toList()));
+
+                            J.Lambda lambda = new J.Lambda(Tree.randomId(), Space.build(" ", Collections.emptyList()), Markers.EMPTY, parameters, Space.build(" ", Collections.emptyList()), b.getStatements().size() == 1 ? b.getStatements().get(0).withPrefix(Space.EMPTY) : b, null);
+                            J.Return returnStatement = new J.Return(Tree.randomId(), Space.build("\n", Collections.emptyList()), Markers.EMPTY, lambda);
+                            b = b.withStatements(Collections.singletonList(autoFormat(returnStatement, context)));
```

Comment:
this could be converted to use a java template,  the result is slightly different but functionally equivalent


```java
                            String t = "return (" + ((J.VariableDeclarations)parentMethod.getParameters().get(0)).getVariables().get(0).getName().getSimpleName() + ") -> #{any()};";
                            JavaTemplate template = JavaTemplate.builder(this::getCursor, t).javaParser(() -> JavaParser.fromJavaVersion().build()).doBeforeParseTemplate(System.out::println).build();
                            b = b.withTemplate(template, b.getCoordinates().firstStatement(), b);
                            b = b.withStatements(ListUtils.map(b.getStatements(), (index, stmt) -> {
                                    if (index == 0){
                                        return stmt;
                                    }
                                    return null;
                                }));
```
result:
```java
                    return (web) -> {
                        web.ignoring().antMatchers("/ignore1", "/ignore2");
                    };
```